Pass TurnContext around instead of sub_id

pakrym-oai opened the pull request:
Today `sub_id` is an ID of a single incoming Codex Op submition. We then associate all events triggered by this operation using the same `sub_id`.

At the same time we are also creating a TurnContext per submission and we'd like to start associating some events (item added/item completed) with an entire turn instead of just the operation that started it.

Using turn context when sending events give us flexibility to change notification scheme.

pakrym-oai committed:
Pass TurnContext around instead of sub_id

pakrym-oai committed:
core/tools/events: refactor event emission to use send_event instead of send_event_raw

tools/handlers/mcp_resource: switch to send_event for MCP tool call events

tools/handlers/view_image: use send_event for ViewImageToolCall event

core/tasks/mod: remove stray whitespace

pakrym-oai committed:
core: remove unused import of Event from protocol in tool modules

jif-oai reviewed (approved):
OOC, you sometimes use `&turn_context` and sometimes `turn_context.as_ref()`. Is it on purpose? (if this is not an `Option`)

pakrym-oai committed:
core: replace use of as_ref() with ref in send_event calls

This commit updates calls to sess.send_event to use &context instead of context.as_ref() throughout codex.rs and compact.rs, simplifying argument passing and improving code consistency.
